src/libraries/System.Memory/src/Resources/Strings.resx

```diff
@@ -123,8 +123,8 @@
   <data name="NotSupported_CannotCallGetHashCodeOnSpan" xml:space="preserve">
     <value>GetHashCode() on Span and ReadOnlySpan is not supported.</value>
   </data>
-  <data name="Argument_InvalidTypeWithPointersNotSupported" xml:space="preserve">
-    <value>Cannot use type '{0}'. Only value types without pointers or references are supported.</value>
+  <data name="Argument_NeedValueTypeWithNoRefs" xml:space="preserve">
+    <value>Type '{0}' must be a value type without references.</value>
```

Comment:
I think the structure original error message was better. It suggested that the type cannot be used with the given API.

The new message suggests that users need to go and fix the type to not have references that will be impossible most of the time.

Reply:
I do not think that's accurate. The type is allowed as the type argument, but the API fails when one does so.

My suggestion would be: "The type '{0}' is not supported because it contains references."